src/søknad/steg/4-barnasbosted/OmAndreForelder.tsx

```diff
@@ -15,53 +14,49 @@ interface Props {
   settForelder: Function;
   forelder: IForelder;
 }
+const OmAndreForelder: React.FC<Props> = ( { barn, settForelder, forelder }) => {
+    const intl = useIntl();
+    const [huketAv, settHuketAv] = useState<boolean>(false);
 
-const OmAndreForelder: React.FC<Props> = ({ barn, settForelder, forelder }) => {
-  const intl = useIntl();
+    const hukAv = (e: any) => {
+        settHuketAv(e.target.checked);
+
+        const nyForelder = {...forelder};
+    
+        if (e.target.checked) {
+          delete nyForelder.navn;
+          delete nyForelder.fødselsdato;
+          delete nyForelder.personnr;
+        }
+
+        settForelder(nyForelder);
+    }
 
   return (
     <>
       <KomponentGruppe>
-        <FeltGruppe>
-          <Element>
-            {barn.navn}
-            {intl.formatMessage({ id: 'barnasbosted.element.andreforelder' })}
-          </Element>
-        </FeltGruppe>
-        <FeltGruppe>
-          <Input
-            className="foreldre-navn-input"
-            onChange={(e) =>
-              settForelder({ ...forelder, navn: e.target.value })
-            }
-            label="Navn"
-          />
-        </FeltGruppe>
-      </KomponentGruppe>
-      <KomponentGruppe>
-        <div className="fødselsnummer">
-          <div className="fødselsdato">
-            <Datovelger
-              settDato={(e: Date | null) =>
-                settForelder({ ...forelder, fødselsdato: e })
-              }
-              valgtDato={
-                forelder.fødselsdato ? forelder.fødselsdato : undefined
-              }
-              tekstid={'datovelger.fødselsdato'}
-              datobegrensning={DatoBegrensning.TidligereDatoer}
+    <FeltGruppe>
+        <Input className="foreldre-navn-input" onChange={(e) => settForelder({...forelder, "navn": e.target.value})} value={forelder.navn ? forelder.navn : ""} label="Navn" disabled={huketAv} />
+    </FeltGruppe>
+    </KomponentGruppe>
+    <KomponentGruppe>
+    <div className="fødselsnummer">
+        <div className="fødselsdato">
+    <Datovelger
+                settDato={(e: Date | null) => settForelder({...forelder, "fødselsdato": e})}
+                valgtDato={forelder.fødselsdato ? forelder.fødselsdato : undefined}
+                tekstid={'datovelger.fødselsdato'}
+                datobegrensning={DatoBegrensning.TidligereDatoer}
             />
-          </div>
-          <Input
-            className="personnummer"
-            onChange={(e) =>
-              settForelder({ ...forelder, personnr: e.target.value })
-            }
-            label="Personnummer. Kun hvis barnet har fått."
-          />
+            </div>
+        <Input className="personnummer" onChange={(e) => settForelder({...forelder, "personnr": e.target.value})} value={forelder.personnr ? forelder.personnr : ""} label="Personnummer (hvis du vet)" disabled={huketAv} />
         </div>
```

Comment:
Her kan man kanskje bruke `PersonInfoGruppe`-komponenten. Den inneholder fødselsdato/datovelger og inputfelt for personnummer. 💡 
Kan hende den kan utvides til å ta inn checkbox også eller kan tweakes slik at de er separate 🤷‍